lib.php

```diff
@@ -115,6 +115,24 @@ function local_augmented_get_units() {
     );
 }
 
+function local_augmented_teacher_get_user_enrolments_from_reminders($reminders): array {
+    $course_ids = [];
+
+    foreach ($reminders as $reminder) {
+        if ($cm = get_coursemodule_from_id('', $reminder->cmid)) {
+            $course_ids[] = $cm->course;
+        }
+    }
+
+    $user_enrolments_repo = \local_augmented_teacher\user_enrolments_factory::get_repository();
+
+    if (empty($course_ids)) {
+        return [];
+    }
+
+    return $user_enrolments_repo->get_by_course_ids($course_ids);
+}
```

Comment:
You could move this method to the user enrolments repo and call it "get_from_reminders()". Then add the $reminders_repo as the second param to the constructor of the user enrolments repo. This would allow you to call $this->reminders_repo->get_all() to get all the reminders. This way the $reminders param could be removed and it would be easier to test :-)